src/settings.c

```diff
@@ -138,31 +153,32 @@ settings_handle_read (XdpSettings           *object,
   g_debug ("Read %s %s", arg_namespace, arg_key);
 
   // TODO: Handle kdeglobals via same interface
-  if (!g_hash_table_contains (self->settings, arg_namespace))
+  if (strcmp (arg_namespace, "org.gnome.fontconfig") == 0)
     {
-      g_debug ("Attempted to read from unknown namespace %s", arg_namespace);
-      g_dbus_method_invocation_return_error_literal (invocation, XDG_DESKTOP_PORTAL_ERROR,
-                                                     XDG_DESKTOP_PORTAL_ERROR_NOT_FOUND,
-                                                     _("Requested setting not found"));
+      if (strcmp (arg_key, "timestamp") == 0)
+        {
+          g_dbus_method_invocation_return_value (invocation,
+                                                 g_variant_new ("(v)", g_variant_new_int32 (self->fontconfig_timestamp)));
+          return TRUE;
+        }
     }
-  else
+  else if (g_hash_table_contains (self->settings, arg_namespace))
     {
       SettingsBundle *bundle = g_hash_table_lookup (self->settings, arg_namespace);
       if (g_settings_schema_has_key (bundle->schema, arg_key))
         {
           g_autoptr (GVariant) variant = NULL;
           variant = g_settings_get_value (bundle->settings, arg_key);
-          g_dbus_method_invocation_return_value (invocation, g_variant_new("(v)", variant));
-        }
-      else
-        {
-          g_debug ("Attempted to read unknown key %s", arg_key);
-          g_dbus_method_invocation_return_error_literal (invocation, XDG_DESKTOP_PORTAL_ERROR,
-                                                         XDG_DESKTOP_PORTAL_ERROR_NOT_FOUND,
-                                                         _("Requested setting not found"));
+          g_dbus_method_invocation_return_value (invocation, g_variant_new ("(v)", variant));
+          return TRUE;
         }
     }
 
```

Comment:
I've turned around the logic here to use early returns for successful cases, and have a catch-all error case at the end.